### DISCUSSION TITLE
Generation of _soc_inthandlers.h for Zephyr

### DISCUSSION THREAD

**Discussion by sunilkumardommati** [thesofproject/sof#7849] [General]
Hi,

We are working on porting Zephyr to our platform.
We are not sure on how to generate the **_soc_inthandlers.h** file which is placed  at **zephyr\soc\xtensa\soc_vendor\\soc\include** path.

I tried to generate using, **zephyr\arch\xtensa\core\xtensa_intgen.py**  with our platform specific **core-isa.h** as input.
But its only producing the following code but not complete interrupt handlers. 

`/*
* THIS FILE WAS AUTOMATICALLY GENERATED.  DO NOT EDIT.
*
* Functions here are designed to produce efficient code to
* search an Xtensa bitmask of interrupts, inspecting only those bits
* declared to be associated with a given interrupt level.  Each
* dispatcher will handle exactly one flagged interrupt, in numerical
* order (low bits first) and will return a mask of that bit that can
* then be cleared by the calling code.  Unrecognized bits for the
* level will invoke an error handler.
*/

#include <xtensa/config/core-isa.h>
#include <zephyr/sys/util.h>
#include <zephyr/sw_isr_table.h>

static inline int _xtensa_handle_one_int0(unsigned int mask)
{
        return 0;
}
`

please let me know if I’m missing anything here.

**Comment by lgirdwood** [comment]
@sunilkumardommati you may want to ask on Zephyr project, but @andyross may be able to help you here too.

**Comment by sunilkumardommati** [comment]
@andyross, could you please let us know your inputs on this issue?

**Reply by lgirdwood** [reply]
Also check the Cmakefile.txt as IIUC it invokes the generation of this file.

**Comment by sunilkumardommati** [comment]
@lgirdwood , we've got inputs in Zephyr devel forum that we need to run the xtensa_intgen.tmpl through the toolchain's preprocessor, and then use xtensa_intgen.py to process the output.
With this, we are able to generate the int handlers file.
